get_http.go

```diff
@@ -117,13 +118,54 @@ func (g *HttpGetter) GetFile(dst string, u *url.URL) error {
 		g.Client = httpClient
 	}
 
-	resp, err := g.Client.Get(u.String())
+	// check to see whether user has specified a range of bytes to download.
+	// if user has, but the range is invalid, fall back to downloading the
+	// whole file
+	byteRange, rangeErr := GetByteRange(u)
+	if rangeErr != nil {
+		if strings.Compare("No byte range provided", rangeErr.Error()) != 0 {
+			fmt.Printf("%s; going to download entire file without a range request",
+				rangeErr)
+		}
+	}
+
+	req, err := http.NewRequest("HEAD", u.String(), nil)
+
+	isPartialDownload := false
+	if rangeErr == nil {
+		// We first make a HEAD request so we can check if the server supports
+		// range queries. If the server/URL doesn't support HEAD requests,
+		// we just fall back to GET.
+
+		if err != nil {
+			return err
+		}
+
+		resp, err := g.Client.Do(req)
+		if err == nil && (resp.StatusCode >= 200 && resp.StatusCode < 300) {
+			// If the HEAD request succeeded, then attempt to set the range
+			// query if we can.
+			if resp.Header.Get("Accept-Ranges") == "bytes" {
+				req.Header.Set("Range", fmt.Sprintf("bytes=%s-%s",
+					byteRange[0], byteRange[1]))
+				isPartialDownload = true
+			}
+		}
```

Comment:
What is the HEAD request failed? What if `err != nil` here? If we intend to ignore errors and download the full file, we should probably at least log that the HEAD request failed so range request will be ignored.